src/main/java/teammates/ui/webapi/action/DeleteInstructorAction.java

```diff
@@ -31,9 +40,60 @@ public ActionResult execute() {
         String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
 
         InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, instructorId);
+        if (instructor == null) {
+            CourseAttributes courseAttributes = logic.getCourse(courseId);
+            if (courseAttributes == null) {
+                return new JsonResult("No course with id: " + courseId, HttpStatus.SC_NOT_FOUND);
+            }
+            return new JsonResult("No instructor with id: " + instructorId, HttpStatus.SC_NOT_FOUND);
+        }
+
+        // Deleting last instructor from the course is not allowed if you're not the admin
+        if (userInfo.isInstructor && !hasAlternativeInstructor(courseId, instructor.email)) {
+            return new JsonResult("The instructor you are trying to delete is the last instructor in the course. "
+                    + "Deleting the last instructor from the course is not allowed.", HttpStatus.SC_BAD_REQUEST);
+        }
+
         logic.deleteInstructor(courseId, instructor.email);
 
-        return new JsonResult("Instructor is successfully deleted.", HttpStatus.SC_OK);
+        return new JsonResult("OK", HttpStatus.SC_OK);
```

Comment:
I agree with @rrtheonlyone that we should make messages more specific.

Reply:
Because of [this](https://github.com/TEAMMATES/teammates/pull/9437#discussion_r260258457). I talked with Xiao Pu and he said that delete should just return "OK", so I'm following convention with Course. Note that "Instructor is successfully deleted." was the previous message, and the message to be shown to the user, but the API shouldn't concern itself with such things - it's the frontend that should care what is shown.

Reply:
I didn't say it should be `OK` 😕 . Any reasonable message is fine.

> but the API shouldn't concern itself with such things

This does not prevent you from using reasonable/meaningful message. The frontend just won't use the message from the backend anyway.